solutions/usaco-994.mdx

```diff
@@ -2,72 +2,110 @@
 id: usaco-994
 source: USACO Gold 2020 January
 title: Farmer John Solves 3SUM
-author: Qi Wang
+author: Qi Wang, Ryan Chou
 ---
 
 [Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_threesum_gold_jan20.html)
 
+## Explanation
+
+Although the problem may be daunting at first, it helps to break down the problem into separate components.
+
+Let's use a DP!
+
+If $\texttt{ways}[i][j]$ equals the number of unordered triplets which sum to zero between $[i, j]$, then what states does this rely on?
+
+![](usaco-994/init.png)
+
+Well intuitively, we can choose to exclude one point or the other, which gives us a picture like this (reminder to make sure not to double count the overlap).
+
+![](usaco-994/ps.png)
+
+However, we're missing something! We aren't taking account for the for any triplets which start at $i$ and end at $j$, since those aren't included in our "PIE-like" transition.
```

Comment:
```suggestion
However, we're missing something! We aren't taking into account any triplets which start at $i$ and end at $j$, since those aren't included in our "PIE-like" transition.